aspnetcore/security/enforcing-ssl.md

```diff
@@ -39,3 +59,34 @@ For more information, see [URL Rewriting Middleware](xref:fundamentals/url-rewri
 
 Requiring HTTPS globally (`options.Filters.Add(new RequireHttpsAttribute());`) is a security best practice. Applying the
 `[RequireHttps]` attribute to all controllers/Razor Pages isn't considered as secure as requiring HTTPS globally. You can't guarantee the `[RequireHttps]` attribute is applied when new controllers and Razor Pages are added.
+
+-------------------
+
+<a name="hsts"></a>
+## HTTP Strict Transport Security Protocol (HSTS)
+
+
+Per [OWASP](https://www.owasp.org/index.php/About_The_Open_Web_Application_Security_Project), [HTTP Strict Transport Security (HSTS)](https://www.owasp.org/index.php/HTTP_Strict_Transport_Security_Cheat_Sheet) is an opt-in security enhancement that is specified by a web application through the use of a special response header. Once a supported browser receives this header that browser will prevent any communications from being sent over HTTP to the specified domain and will instead send all communications over HTTPS. It also prevents HTTPS click through prompts on browsers.
+
+ASP.NET Core 2.1 preview1 or later implements HSTS with the `UseHsts` extension method. The following code calls `UseHsts` when the app isn't in [development mode](xref:fundamentals/environments):
+
+[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet1&highlight=10)]
+
+`UseHsts` not recommend in development because calling `UseHsts` excludes the local loopback address.
+
+The following code:
+
+[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=11-16)]
+
+* Sets the preload parameter of the Strict-Transport-Security header. Preload is not part of the [RFC HSTS specification](https://tools.ietf.org/html/rfc6797), but is supported by web browsers to preload HSTS sites on fresh install. See [https://hstspreload.org/](https://hstspreload.org/) for more information.
+* Enables [includeSubDomain](https://tools.ietf.org/html/rfc6797#section-6.1.2), which applies the HSTS policy to Host subdomains. 
+* Explicitly sets the max-age parameter of the Strict-Transport-Security header to to 60 days. If not set, defaults to 30 days. See the [max-age directive](https://tools.ietf.org/html/rfc6797#section-6.1.1) for more information.
+
+`UseHsts` excludes the following loopback hosts:
+
+* `localhost` : The IPv4 loopback address.
+* `127.0.0.1` : The IPv4 loopback address.
+* `[::1]` : The IPv6 loopback address.
```

Comment:
Add that you also may specify other excluded hosts on the HstsOptions.

Reply:
No setter
See [source](https://github.com/aspnet/BasicMiddleware/blob/release/2.1/src/Microsoft.AspNetCore.HttpsPolicy/HstsOptions.cs#L43)

Reply:
You can still call options.ExcludedHosts.Add(...) to add onto the list.

Reply:
For some reason I can't get the version that has `ExcludedHosts`
I'm using `<PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.0-preview2-final" />`

![image](https://user-images.githubusercontent.com/3605364/38706498-10a4c9ec-3e49-11e8-88eb-2d0084e63def.png)

Reply:
It is part of the HstsOptions, not the HttpsRedirectionOptions.

Reply:
Can you send me the code to do that?

Reply:
Never mind, wrong place.

Reply:
```
 services.AddHsts(options =>
 {
     options.Preload = true;
     options.IncludeSubDomains = true;
     options.MaxAge = TimeSpan.FromDays(60);
     options.ExcludedHosts.Add("example.com");
     options.ExcludedHosts.Add("www.example.com");
 });
```

Reply:
Yeah looks good!